### ISSUE TITLE
[rootlesskit:child ] error: executing [[ip tuntap add name tap0 mode tap]: exit status 1

### ISSUE THREAD

**Issue by kenorb** [rootless-containers/rootlesskit#181]
```
+ exec rootlesskit --net=vpnkit --mtu=1500 --slirp4netns-sandbox=auto --slirp4netns-seccomp=auto --disable-host-loopback --port-driver=builtin --copy-up=/etc --copy-up=/run /opt/docker/bin/dockerd-rootless.sh --debug --experimental --iptables=false --storage-driver vfs
WARN[0000] failed to mount sysfs, falling back to read-only mount: operation not permitted 
WARN[0000] failed to mount sysfs: operation not permitted 
open: No such file or directory
[rootlesskit:child ] error: executing [[ip tuntap add name tap0 mode tap] [ip link set tap0 address 02:50:00:00:00:01]]: exit status 1
```

**Comment by AkihiroSuda** [comment]
From your other issues I suspect this is related to Colab.

I don’t even know what is Colab, but this is something needs to be fixed on Colab side. Probably they have to relax seccomp confinement and/or LSM like AppArmor.

**Comment by kenorb** [comment]
> From your other issues I suspect this is related to Colab.
> 
> I don’t even know what is Colab, but this is something needs to be fixed on Colab side. Probably they have to relax seccomp confinement and/or relax LSM like AppArmor.

I'm using rootless Docker by installing this script: <https://get.docker.com/rootless> which installing [dockerd-rootless.sh](https://github.com/docker/docker-ce/blob/master/components/engine/contrib/dockerd-rootless.sh) wrapper script which is using rootlesskit which suppose to be rootless, but it actually doesn't work without the root, so that's what this bug is about. Related issue: <https://github.com/docker/for-linux/issues/1124>.

Colab is just the collaborative VM. The demo of this problem can be found at:

- https://colab.research.google.com/drive/1oRja4v-PtY6lFMJIIF79No4s3s-vbqd4

Basically when some network parameters are passed like `--net=vpnkit --mtu=1500 --slirp4netns-sandbox=auto --slirp4netns-seccomp=auto`, it fails due to lack of permissions.

**Comment by kenorb** [comment]
Simpler test to reproduce the issue:

```
%%shell
curl -s https://download.docker.com/linux/static/stable/x86_64/docker-19.03.9.tgz | tar vxz --strip=1 -C /usr/local/bin/
curl -s https://download.docker.com/linux/static/stable/x86_64/docker-rootless-extras-19.03.9.tgz | tar vxz --strip=1 -C /usr/local/bin/
useradd -md /home/docker docker
apt-get -qqq install iproute2 uidmap
docker --version
```

```
%%shell
sudo -Hu docker dockerd-rootless.sh
```

Output:

```
WARN[0000] failed to mount sysfs, falling back to read-only mount: operation not permitted 
WARN[0000] failed to mount sysfs: operation not permitted 
open: No such file or directory
[rootlesskit:child ] error: executing [[ip tuntap add name tap0 mode tap] [ip link set tap0 address 02:50:00:00:00:01]]: exit status 1
```

**Comment by AkihiroSuda** [comment]
Colab is already running inside Docker without --privileged.

Running rootless Docker (and rootful Docker of course) inside Docker is possible only when the parent Docker container is running with --privileged.

**Comment by AkihiroSuda** [comment]
BTW Docker-in-UML-in-Docker may work https://github.com/weber-software/diuid
